packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -673,11 +515,35 @@ function createArrayNodeProxy(
 	return proxy;
 }
 
-abstract class CustomArrayNodeBase<const T extends ImplicitAllowedTypes> extends TreeNodeValid<
-	Iterable<InsertableTreeNodeFromImplicitAllowedTypes<T>>
-> {
+type Insertable<T extends ImplicitAllowedTypes> = readonly (
+	| InsertableTreeNodeFromImplicitAllowedTypes<T>
+	| IterableTreeArrayContent<InsertableTreeNodeFromImplicitAllowedTypes<T>>
+)[];
+
+abstract class CustomArrayNodeBase<const T extends ImplicitAllowedTypes>
+	extends TreeNodeWithArrayFeatures<T>
+	implements TreeArrayNode<T>
```

Comment:
This implements clause is the key improvement for this PR: the implementation and the API now actually have type checking that they align.

Reply:
Thank you for pointing this out, it makes this all much easier to review!